contrib/platform/src/com/sun/jna/platform/win32/COM/util/ProxyObject.java

```diff
@@ -513,10 +550,83 @@ private void setProperty(DISPID dispID, Object... args) {
         String iidStr = annotation.iid();
+        String nameStr = annotation.name();
         if (null != iidStr && !iidStr.isEmpty()) {
             return new IID(iidStr);
+        } else if (null != nameStr && !nameStr.isEmpty()) {
+            ITypeInfo typeInfo;
+
+            try {
+                typeInfo = ProxyObject.this.getTypeInfo();
+            } catch (COMException e) {
+                throw new COMException("failed to read type infos to resolve the interface name.", e);
+            }
+
+            if (typeInfo == null) {
+                throw new COMException("no type infos provided to resolve the interface name.");
+            }
+
+            String[] allImplemented = null;
+
+            try {
+                TypeInfoUtil typeInfoUtil = new TypeInfoUtil(typeInfo);
+
+                OaIdl.TYPEATTR typeAttr = typeInfoUtil.getTypeAttr();
+                IID typeIid = new IID(typeAttr.guid);
+                typeInfoUtil.ReleaseTypeAttr(typeAttr);
+
+                int nImplemented = typeAttr.cImplTypes.intValue();
+                allImplemented = new String[nImplemented + 1];
+
+                for (int i = -1; i < nImplemented; i++) {
+                    ITypeInfo info;
+
+                    if (i != -1) {
+                        OaIdl.HREFTYPE refTypeOfImplType = typeInfoUtil.getRefTypeOfImplType(i);
+                        info = typeInfoUtil.getRefTypeInfo(refTypeOfImplType);
+                    } else {
+                        info = typeInfo;
+                    }
+
+                    WTypes.BSTRByReference pBstrName = new WTypes.BSTRByReference();
+                    // MEMBERID_NIL will return the documentation for the type
+                    info.GetDocumentation(OaIdl.MEMBERID_NIL, pBstrName, null, null, null);
+
+                    WTypes.BSTR bstrName = pBstrName.getValue();
+                    String implName = bstrName.toString();
+                    OleAuto.INSTANCE.SysFreeString(bstrName);
+
+                    if (nameStr.equals(implName)) {
+                        if (i != -1) {
+                            // discover the guid if the implemented type
+                            TypeInfoUtil implTypeInfoUtil = new TypeInfoUtil(info);
+                            OaIdl.TYPEATTR implTypeAttr = implTypeInfoUtil.getTypeAttr();
+                            IID iid = new IID(implTypeAttr.guid);
+                            implTypeInfoUtil.ReleaseTypeAttr(implTypeAttr);
+
+                            return iid;
+                        } else {
+                            return typeIid;
+                        }
+                    } else {
+                        allImplemented[i + 1] = implName;
+                    }
+                }
+            } catch (COMException e) {
+                throw new COMException("failed to read implemented type infos to resolve the interface name.", e);
+            }
+
+            if ("IUnknown".equals(nameStr)) {
```

Comment:
I don't understand the special case. In my mind every usage outside dispatch is a huge problem (as already indicated) and highly dangerous. So of course trying to dispatch against `IUnknown` is stupid, but the same is true for every interface, that is not derived from `IDispatch`.